resolver/resolver.go

```diff
@@ -38,19 +45,21 @@ func (e ErrNoLink) Error() string {
 type ResolveOnce func(ctx context.Context, ds ipld.NodeGetter, nd ipld.Node, names []string) (*ipld.Link, []string, error)
 
 // Resolver provides path resolution to IPFS
-// It has a pointer to a DAGService, which is uses to resolve nodes.
+// It has a pointer to a FetcherConfig, which is uses to resolve nodes.
 // TODO: now that this is more modular, try to unify this code with the
 //       the resolvers in namesys
 type Resolver struct {
```

Comment:
I could use more docs on this.  I see there weren't a ton before, either, but if we're gonna lug this into the future, it would be good to have some comments on whether its shape is intentful or primarily legacy driven; any notes about what it *doesn't* do, etc.

Reply:
I'm kinda wondering, throughout this whole file:

What's a `Resolver` for, vs using `traverse.*` on the nodes?  (Is it just legacy API satisfying to try to reduce change splash size?  Fine, but we should mark wherever that reasoning is used; that information failing to transfer to the next person working in this area would be devastating to that person's ability to plan changes.)

Is `FetcherConfig` adding any key data vs `traversal.Config`?  Or did we accidentally create a similar abstraction twice, and we should try to simplify it back to one?  If there is key data it adds that makes it suitable for `Resolver` -- what is it?

Reply:
Should we be trying to push a lot harder on the ADL'ization of pathing over unixfsv1 stuff?  If we do so, can we make this `Resolver` thing a _much_ thinner facade over that?